packages/db/src/indexes/index-registry.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Index Dev Mode - Helps developers identify when indexes would improve performance
+ *
+ * Dev mode suggestions are ON by default in non-production builds.
+ */
+
+// Dev mode detection settings - ON by default in non-production
+let devModeConfig: IndexDevModeConfig = {
+  enabled: true,
+  collectionSizeThreshold: 1000,
```

Comment:
This threshold seems arbitrary. Going through an non-indexed collection of 1000 rows should be pretty fast. I guess modern machines can easily go to perhaps 1M rows before it really becomes slow and needs indexes. Time-based suggestions (`slowQueryThresholdMs`) makes much more sense imo.